raspberry/bzzz/sensors/pressure_sensor.py

```diff
@@ -1,210 +1,185 @@
+        av = self.__initialise()
+        self.__av_pressure = av
 
-    def __get_measurements_in_background_t(self, DEVICE_ADDRESS, reference_pressure_at_sea_level):
-        while True:
-            for _ in range(self.window_length):
-                self.__pressure_readings_list.append(self._get_current_pressure_measurement())
-            if self.reference_pressure_at_sea_level is None:
-                self.reference_pressure_at_sea_level = self.data_processor.process(self.__pressure_readings_list)
-        
-            self.update_altitude()
-        # Write data into the log file
-            if self.log_file is not None:
-        
+    def __pressure_correction(self, p):
+        delta = self.__av_pressure - self.__reference_pressure_at_sea_level
+        return p - delta
+    
+    def __get_measurements_in_background(self):
+        while self.__keep_running:
+            pres_raw, temp = self.__pressure_temp_from_sensor()
+            pres = self.__pressure_correction(pres_raw) if self.__av_pressure > 0 else pres_raw
+            alt = _calculate_altitude(pres, temp, self.__reference_pressure_at_sea_level) 
+            self.__values_cache[self.__cursor, :] = np.array([pres, alt])
+            self.__cursor = (self.__cursor + 1) % self.__window_length
+            if self.__log_file is not None and self.__av_pressure > 0:
                 current_timestamp = datetime.datetime.now()
-                data_to_log = (self.pressure, self.altitude)
-                self.logger.record(current_timestamp, data_to_log)
-                    
-
-    def __readBmp180Id(self):
-        REG_ID = 0xD0
-        chip_id, chip_version = self.smbus.read_i2c_block_data(self.DEVICE_ADDRESS,
-                                                                REG_ID, 2)
-        return chip_id, chip_version  
-         
-    def __readBmp180(self):
-        # Register Addresses
-        REG_CALIB = 0xAA
-        REG_MEAS = 0xF4
-        REG_MSB = 0xF6
-        REG_LSB = 0xF7
-    # Control Register Address
-        CRV_TEMP = 0x2E
-        CRV_PRES = 0x34
-    # Oversample setting
-        OVERSAMPLE = 3    # 0 - 3
-
-    # Read calibration data
-    # Read calibration data from EEPROM
-        cal = self.smbus.read_i2c_block_data(self.DEVICE_ADDRESS,
-                                                REG_CALIB, 22)
-
-    # Convert byte data to word values
-        AC1 = _getShort(cal, 0)
-        AC2 = _getShort(cal, 2)
-        AC3 = _getShort(cal, 4)
-        AC4 = _getUshort(cal, 6)
-        AC5 = _getUshort(cal, 8)
-        AC6 = _getUshort(cal, 10)
-        B1 = _getShort(cal, 12)
-        B2 = _getShort(cal, 14)
-        MB = _getShort(cal, 16)
-        MC = _getShort(cal, 18)
-        MD = _getShort(cal, 20)
+                data_to_log = (pres, alt)
+                self.__logger.record(current_timestamp, data_to_log)  
 
-    # Read temperature
-        self.smbus.write_byte_data(self.DEVICE_ADDRESS, REG_MEAS, CRV_TEMP)
+    def pressure(self):
```

Comment:
In anemometer, the corresponding getters are annotated with `@property`; we should do the same everywhere for the sake of uniformity.